### PR TITLE
Ingest content placeholders

### PR DESCRIPTION
**Pull request by TheoLeanse**
@i-like-robots @adgad

### CODE REVIEW

File: server/data-sources/types/article.js [hunk 1]
```diff
@@ -10,9 +10,16 @@ const isFastFT = (content) => (
...
+	content &&
+	content.webUrl &&
+	!content.webUrl.match(/^https:\/\/www.ft.com/) &&
+	content.types.find(type => type === 'http://www.ft.com/ontology/content/Content')
+);
+
 class ArticleType extends Type {
 	static match ({ capiV1, capiV2 }) {
-		return isCapiV1(capiV1) || isFastFT(capiV2);
+		return isCapiV1(capiV1) || isFastFT(capiV2) || isContentPlaceholder(capiV2);
```
Comment (i-like-robots) [inline server/data-sources/types/article.js]: Should this be a new content type rather than an article?

File: server/lib/elastic/model.js [hunk 2]
```diff
@@ -3,16 +3,23 @@ const sanityCheck = require('../sanity-check');
...
+	let type;
 	return dataSources(uuid)
+		.then((dataType) => type = dataType)
 		.then((dataType) => {
 			return dataType.model();
 		})
 		.then((result) => {
 			// validate the content before attempting to ingest it
-			const issues = sanityCheck(result);
+			const typeValidationIssues = type.validate && type.validate(result);
```
Comment (adgad) [inline server/lib/elastic/model.js]: Could probably simplify this into something like:
```
const issues = type.validate ? type.validate(result) : sanityCheck(result);
if(issues) {

}
```

File: server/models/placeholder/main-image-v2.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const logger = require('@financial-times/n-logger').default;
```
Comment (adgad) [inline server/models/placeholder/main-image-v2.js]: Is there a reason this file is different to the other main-image-v2 in lib?

File: server/lib/elastic/model.js (hunk 2 above)
Comment (TheoLeanse) [inline server/lib/elastic/model.js]: 👌

File: server/models/placeholder/main-image-v2.js (hunk 3 above)
Comment (TheoLeanse) [inline server/models/placeholder/main-image-v2.js]: only that the alternative images which a placeholder gets has no title or description

**Comment by adgad** [comment]
@TheoLeanse @i-like-robots @kavanagh I _think_ this is good to go now, if y'all approve 😁

File: server/models/placeholder/main-image-v2.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+		ratio: image.pixelWidth && image.pixelHeight ? image.pixelWidth / image.pixelHeight : null
+	}))
+
+	.catch((err) => {
+		logger.error({
+			event: 'MAIN_IMAGE_FETCH_FAIL',
+			error: err.toString(),
+			uuid: imageId
+		});
+	})
```
Comment (adgad) [inline server/models/placeholder/main-image-v2.js]: I think it's slightly different to the `main-image-v2` in article - in that it's not a mainImage and is not an imageSet. But they should probably be refactored to share the same code.

Comment (adgad) [inline server/models/placeholder/main-image-v2.js]: I'm going to rename to alternative-image-v2 for now so it's less confusing - but images will need a bit of a redo when we come to putting the topper in.

**Comment by i-like-robots** [comment]
👍

**Pull request merged**